### PR TITLE
Multibyte char alternative to #29

### PR DESCRIPTION
**Pull request by Fabs**
An attempt to solve #29 without such a big hit on performance (and less code too luckily).

```
Original x 4,945,476 ops/sec ±1.07% (83 runs sampled)
First Solution x 63,137 ops/sec ±1.72% (79 runs sampled)
New Solution x 1,773,737 ops/sec ±1.44% (88 runs sampled)
Fastest is Original
```
I did a small benchmark to show a point, to prove we need a best one. Instead of 80x slower then the original solution we are being jus 3x here. 

This solution also do not need the extra package `text-encoding` that I left so I could bechmark.

You can edit `benchmark.js` and run with:
`node benchmark.js`

### CODE REVIEW

**Comment by jaxesn** [comment]
This answer might be worth looking at, looks like there might be an edge case you missed.

https://stackoverflow.com/a/23329386

**Comment by juliangieseke** [comment]
Jackson is right, even with that answer I wasnt able to get it working with emojis:
```
function countExtraCodepoints(string, start, end) {
  let extraCodePoints = 0;

  for (let i = 0; i < end - 1; i++) {
    let codePoint = string.charCodeAt(start + i);

    if (codePoint >= 0x80 && codePoint <= 0x7ff) extraCodePoints++;

    if (
      (codePoint >= 0x800 && codePoint <= 0xd7ff) ||
      (codePoint >= 0xe000 && codePoint <= 0xffff)
    )
      extraCodePoints += 2;

    if (codePoint >= 0x10000 && codePoint <= 0x10ffff) extraCodePoints += 3;
  
    if (codePoint >= 0xdc00 && codePoint <= 0xdfff) extraCodePoints--; //trail surrogate?!

    i++;
  }

  return extraCodePoints;
}
```

This Testcase still fails (it is green with my solution):
```
  it("returns multiple records with utf8 in many messages (emoji)", function() {
    expect(read("6\n1234ß5\nß23414\n🤷‍♂️.")).toEqual([
      ["1234ß", "ß234", "🤷‍♂️."],
      ""
    ]);
  });
```

**Comment by Fabs** [comment]
I fixed the implementation again. I added the test that was broken.

**Comment by Fabs** [comment]
Yes @jaxesn. I have not considered codepoints bigger than 2 bytes (I was just making it simpler). I used that article to help me fix that for all utf-8. Thanks @jaxesn and @juliangieseke

**Comment by Fabs** [comment]
I would ask @orlandohohmeier to share his benchmark results. It looks that this solution is suitable time wise, and I think now it is correct, although I would like to see an integration test on dcos-ui for a real json with and without UTF-8 larger characters.

File: src/read.js [hunk 1]
```diff
@@ -4,13 +4,29 @@ var copychars = require("@dcos/copychars").default;
 
 var RECORD_PATTERN = /^\d+\n.+/;
 
+function countExtraCodepoints(string, start, end) {
+  let correctedLength = end - start;
```
Comment (jaxesn) [inline src/read.js]: super minor, but what do you think of this


```
  let length = end - start;
  let correctedLength = length;
  for (let i = length; i >= start; i--) {
```

Comment (Fabs) [reply src/read.js]: Yes. I did not write it with legibility in mind, there is this and a few more I would like to fix if we are going to go for this solution. I also have to lint it and apply the same rigor we do with code on DC/OS.

**Comment by orlandohohmeier** [comment]
Closed in favor of https://github.com/dcos-labs/recordio/pull/31

**Pull request closed without merging**